d2lib/d2.go

```diff
@@ -68,14 +68,27 @@ func compile(ctx context.Context, g *d2graph.Graph, opts *CompileOptions) (*d2ta
 			return nil, err
 		}
 
-		constantNears := d2near.WithoutConstantNears(ctx, g)
+		constantNearGraphs := d2near.WithoutConstantNears(ctx, g)
+
+		// run core layout for constantNears
+		for _, tempGraph := range constantNearGraphs {
+			nearObject := tempGraph.Root.ChildrenArray[0]
+			nearObject.Parent = tempGraph.Root
+			if err = coreLayout(ctx, tempGraph); err != nil {
+				return nil, err
+			}
+			nearObject.Parent = g.Root
+			for _, obj := range tempGraph.Objects {
+				obj.Graph = g
+			}
```

Comment:
and then this belongs in `d2near.Layout`

Reply:
fixed

Reply:
is it? i still see it here

Reply:
done...